src/Microsoft.AspNetCore.Server.IISIntegration/NativeMethods.cs

```diff
@@ -62,18 +63,33 @@ public enum REQUEST_NOTIFICATION_STATUS
         public unsafe static extern void http_set_response_status_code(IntPtr pHttpContext, ushort statusCode, byte* pszReason);
 
         [DllImport(AspNetCoreModuleDll)]
-        public unsafe static extern int http_read_request_bytes(IntPtr pHttpContext, byte* pvBuffer, int cbBuffer, PFN_ASYNC_COMPLETION pfnCompletionCallback, IntPtr pvCompletionContext, out int dwBytesReceived, out bool fCompletionExpected);
+        public unsafe static extern int http_read_request_bytes(IntPtr pHttpContext, byte* pvBuffer, int cbBuffer, out int dwBytesReceived, out bool fCompletionExpected);
 
         [DllImport(AspNetCoreModuleDll)]
         public unsafe static extern bool http_get_completion_info(IntPtr pCompletionInfo, out int cbBytes, out int hr);
 
         [DllImport(AspNetCoreModuleDll)]
+        public unsafe static extern bool http_set_managed_context(IntPtr pHttpContext, IntPtr pvManagedContext);
+
+        [DllImport(AspNetCoreModuleDll)]
         [return: MarshalAs(UnmanagedType.BStr)]
         public unsafe static extern string http_get_application_full_path();
 
         [DllImport(AspNetCoreModuleDll)]
         public unsafe static extern bool http_shutdown();
 
+        [DllImport(AspNetCoreModuleDll)]
+        public unsafe static extern int http_websockets_read_bytes(IntPtr pHttpContext, byte* pvBuffer, int cbBuffer, PFN_ASYNC_COMPLETION pfnCompletionCallback, IntPtr pvCompletionContext, out int dwBytesReceived, out bool fCompletionExpected);
+
+        [DllImport(AspNetCoreModuleDll)]
+        internal unsafe static extern int http_websockets_write_bytes(IntPtr pHttpContext, HttpApiTypes.HTTP_DATA_CHUNK* pDataChunks, int nChunks, PFN_ASYNC_COMPLETION pfnCompletionCallback, IntPtr pvCompletionContext, out bool fCompletionExpected);
+
+        [DllImport(AspNetCoreModuleDll)]
+        public unsafe static extern int http_enable_websockets(IntPtr pHttpContext);
+
+        [DllImport(AspNetCoreModuleDll)]
+        public unsafe static extern int http_cancel_io(IntPtr pHttpContext);
```

Comment:
Will be used for canceling pending async reads/writes.

Reply:
Not sure we need this but I guess it doesn't hurt.